crates/astria-sequencer/src/authority/component.rs

```diff
@@ -0,0 +1,75 @@
+use std::sync::Arc;
+
+use anyhow::{
+    Context,
+    Result,
+};
+use tendermint::{
+    abci::request::{
+        BeginBlock,
+        EndBlock,
+    },
+    validator,
+};
+use tracing::instrument;
+
+use super::state_ext::{
+    StateReadExt,
+    StateWriteExt,
+    ValidatorSet,
+};
+use crate::{
+    component::Component,
+    genesis::GenesisState,
+};
+
+#[derive(Default)]
+pub(crate) struct AuthorityComponent;
+
+#[async_trait::async_trait]
+impl Component for AuthorityComponent {
+    type AppState = (GenesisState, Vec<validator::Update>);
+
+    #[instrument(name = "AuthorityComponent:init_chain", skip(state))]
+    async fn init_chain<S: StateWriteExt>(mut state: S, app_state: &Self::AppState) -> Result<()> {
+        // set sudo key and initial validator set
+        state
+            .put_sudo_address(app_state.0.authority_sudo_key)
+            .context("failed to set sudo key")?;
+        state
+            .put_validator_set(ValidatorSet(app_state.1.clone()))
```

Comment:
I don't think it has to be? `AppState` is defined in this crate and the type is specified in this impl, so probably not hard to make it a struct instead of a tuple.